sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/GpuBroadcastExchangeExec.scala

```diff
@@ -50,21 +50,23 @@ class SerializeConcatHostBuffersDeserializeBatch(
   extends Serializable with Arm with AutoCloseable {
   @transient private val headers = data.map(_.header)
   @transient private val buffers = data.map(_.buffer)
-  @transient private var batchInternal: ColumnarBatch = null
+  @transient @volatile private var batchInternalRef: ColumnarBatch = _
 
-  def batch: ColumnarBatch = {
-    if (batchInternal == null) {
+  def batch: ColumnarBatch = this.synchronized {
+    if (batchInternalRef == null) {
       // TODO we should come up with a better way for this to happen directly...
       val out = new ByteArrayOutputStream()
       val oout = new ObjectOutputStream(out)
       writeObject(oout)
       val barr = out.toByteArray
       val oin = new ObjectInputStream(new ByteArrayInputStream(barr))
-      readObject(oin)
+      batchInternalRef = readObject(oin)
     }
-    batchInternal
+    batchInternalRef
   }
 
+  def batchInternal = this.synchronized { batchInternalRef }
```

Comment:
There's no point in synchronizing access to a volatile member.  There's no harm in leaving `batchInternal` a `volatile` and just synchronize access to the `batch` method so two of them don't try to set it at the same time.  All other references are just reads.